### PR TITLE
feat: add push provider api

### PR DESCRIPTION
**Pull request by peterdeme**
Add push provider endpoints

### CODE REVIEW

File: src/Models/Device.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 namespace StreamChat.Models
 {
-    public static class PushProvider
+    public static class PushProviderNames
```
Comment (ferhatelmas) [inline src/Models/Device.cs:7]: in strict sense, it should be `Types` and also it's a breaking change

File: src/Models/PushProvider.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        public string ApnAuthKey { get; set; }
+        [JsonProperty("apn_key_id")]
+        public string ApnKeyId { get; set; }
+        [JsonProperty("apn_team_id")]
+        public string ApnTeamId { get; set; }
+        [JsonProperty("apn_topic")]
+        public string ApnTopic { get; set; }
+
+        [JsonProperty("firebase_server_key")]
+        public string FirebaseServerKey { get; set; }
```
Comment (ferhatelmas) [inline src/Models/PushProvider.cs:47]: drop me

File: src/Clients/IAppClient.cs [hunk 3]
```diff
@@ -48,6 +48,18 @@ public interface IAppClient
...
+        /// <remarks>https://getstream.io/chat/docs/dotnet-csharp/push_introduction/?language=csharp</remarks>
+        Task<UpsertPushProviderResponse> UpsertPushProviderAsync(PushProviderRequest pushProviderRequest);
+
+        /// <summary>Lists all push providers.</summary>
+        /// <remarks>https://getstream.io/chat/docs/dotnet-csharp/push_introduction/?language=csharp</remarks>
+        Task<ListPushProvidersResponse> ListPushProvidersAsync();
+
+        /// <summary>Deletes a push provider.</summary>
+        /// <remarks>https://getstream.io/chat/docs/dotnet-csharp/push_introduction/?language=csharp</remarks>
+        Task<ApiResponse> DeletePushProviderAsync(string providerType, string name);
```
Comment (ferhatelmas) [inline src/Clients/IAppClient.cs:61]: good to type the `Type`

File: src/Models/Device.cs (hunk 1 above)
Comment (peterdeme) [reply src/Models/Device.cs:7]: technically yes but I don't think anyone uses this :D

File: src/Clients/IAppClient.cs (hunk 3 above)
Comment (peterdeme) [reply src/Clients/IAppClient.cs:61]: `type` is a keyword in C#, that's why I went for `providerType`. same for golang

Comment (ferhatelmas) [reply src/Clients/IAppClient.cs:61]: I mean using a type instead of string

File: src/Models/Device.cs (hunk 1 above)
Comment (ferhatelmas) [reply src/Models/Device.cs:7]: most likely

File: src/Clients/IAppClient.cs (hunk 3 above)
Comment (peterdeme) [reply src/Clients/IAppClient.cs]: ✅

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**